File: 155_Min_Stack.cpp
class MinStack {
public:
    vector<int> minStack; 
    MinStack() {
        minStack.clear(); 
    }
    
    void push(int x) {
        minStack.push_back(x); 
    }
    
    void pop() {
        minStack.pop_back(); 
    }
    
    int top() {
        return minStack[ minStack.size() - 1]; 
    }
    
    int getMin() {
        if(minStack.size() >= 0)
        {
            int minn = minStack[0]; 
            for(int c : minStack)
            {
                minn = std::min(minn, c); 
            }
            return minn;
        }
        return -1; 
    }
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack* obj = new MinStack();
 * obj->push(x);
 * obj->pop();
 * int param_3 = obj->top();
 * int param_4 = obj->getMin();
 */

File: 155_Min_Stack.py
class MinStack:
    mystack = None 

    def __init__(self):
        self.mystack =[] 
        
        """
        initialize your data structure here.
        """
        

    def push(self, x: int) -> None:
        self.mystack.append(x) 
        

    def pop(self) -> None:
        self.mystack.pop() 
        

    def top(self) -> int:
        return self.mystack[len(self.mystack) - 1]
        

    def getMin(self) -> int:
        currentMin = float('inf')
        for value in self.mystack: 
            if value < currentMin:
                currentMin = value
        return currentMin 
        


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(x)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()
